src/runtime/components/forms/Textarea.vue

```diff
@@ -157,13 +162,41 @@ export default defineComponent({
       }
     }
 
-    const onInput = (event: InputEvent) => {
-      autoResize()
+    // Custom function to handle the v-model properties
+    const updateInput = (value: string) => {
+      if (props.modelModifiers.trim) {
+        value = value.trim()
+      }
+
+      if (props.modelModifiers.number) {
+        value = looseToNumber(value)
+      }
 
-      emit('update:modelValue', (event.target as HTMLInputElement).value)
+      emit('update:modelValue', value)
       emitFormInput()
     }
 
+    const onInput = (event: InputEvent) => {
+      if (!props.modelModifiers.lazy) {
+        autoResize()
+        updateInput((event.target as HTMLInputElement).value)
+      }
```

Comment:
```suggestion
      autoResize()
      if (!props.modelModifiers.lazy) {
        updateInput((event.target as HTMLInputElement).value)
      }
```

Reply:
@maxsteinwand the autoresize() needs to be triggered independently of the lazy modifier here, otherwise in this case: `<UTextarea autoresize v-model.lazy="text" />` resize will happen only when the user is tabbing out.

Reply:
@romhml fixed it to your suggestion